docs/usage/misc/actions.md

```diff
@@ -0,0 +1,78 @@
+---
+layout: default
+title: Actions
+nav_order: 2
+has_children: false
+parent: Misc
+grand_parent: Usage
+---
+
+# Actions
+
+All OpenHAB's actions including those provided by add-ons are available, notably:
+* Audio
+* Voice
+* Things
+* Ephemeris
+* Exec
+* HTTP
+* Ping
+
+From add-ons, e.g.:
+* Transformation
+* PersistenceExtensions (from Persistence add-on)
+* NotificationAction (from OpenHAB cloud add-on)
+
+For convenience, the following methods are implemented:
+| Method     | Parameters                                                    | Description                                                                                                                          |
+| ---------- | ------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------ |
+| action     | scope, thing_uid                                              | Return the Action object for the given scope and thinguid                                                                            |
+| notify     | msg, email: (optional)                                        | When an email is specified, calls NotificationAction.sendNotification. Otherwise, calls NotificationAction.sendBroadcastNotification |
+| say        | text, volume: (optional), voice: (optional), sink: (optional) | Calls Voice.say()                                                                                                                    |
+| play_sound | filename, volume: (optional), sink: (optional)                | Calls Audio.playSound()                                                                                                              |
+
+## Example
+
+Run the TTS engine and output the default audio sink. For more information see [Voice](https://www.openhab.org/docs/configuration/multimedia.html#voice)
+```ruby
+rule 'Say the time every hour' do
+  every :hour
+  run { say "The time is #{TimeOfDay.now}" }
+end
+```
+
+```ruby
+rule 'Play an audio file' do
+  every :hour
+  run { play_sound "beep.mp3", volume: 100 }
+end
+```
+
+Send a broadcast notification via the OpenHAB Cloud
+```ruby
+rule 'Send an alert' do
+  changed Alarm_Triggered, to: ON
+  run { notify 'Red Alert!' }
+end
+```
+
+Send an email using the Mail binding
+```ruby
+rule 'Send an Email' do
+  every :day
+  run do
+    mail = action('mail', 'mail:smtp:local')
```

Comment:
For the other elements in our library we have items[], things[], etc. 
Should this be:
actions['mail']? 

And I make this comment below, but if actions are attached to things, should it be:
```
things['mail:smtp:local].sendEmail()?
```

I haven't uses actions beyond the basic notify, exec, etc.

Reply:
I'm not sure either actions['mail'] syntax nor things['mail:smtp:local'] syntax would work. Certainly not the things[] syntax.

Perhaps actions[scope][thinguid].methodname_of_the_action could work?

Another example for you (from jython):
```
actions.get("telegram", "telegram:telegramBot:mybotname").sendTelegram(notification_msg)
actions.get("lgwebos", thing.getUID().toString()).launchBrowser('https://www.openhab.org')
actions.get("mqtt", "mqtt:broker:mosquitto").publishMQTT("zigbee/bridge/config/permit_join", "")
```

It is always a pair of `[ bindingname, thingid ]`

Reply:
If I were using multiple calls to the action, I'd do:
```
livingroomtv_actions = actions['lgwebos']['livingroomtv:thing:uid']
livingroomtv_actions.launchBrowser(...)
livingroomtv_actions.dosomethingelse(....)
```

Reply:
Can I play around for a few minutes?  If we don't like it we don't have to use it? I think there might be a way.  If it ends up too messy, we can skip it.  
What would be the best and easiest binding to test an action with?

Reply:
Is there a Ruby construct that lets you have actions[scope][thinguid] translate into $actions.get(scope, thinguid) without having to create two levels of objects?

This is what I can think of with my limited knowledge:
```
        class ActionThingUid
          def initialize(scope)
            @scope = scope
          end

          def [](thing_uid)
            $actions.get(@scope, thing_uid)
          end
        end

        class ActionScope
          def initialize
            @scopes = []
          end

          def [](scope)
            @scopes[scope] = ActionThingUid.new(scope) unless @scopes.key? scope
            @scopes[scope]
          end
        end

        def actions
          @actions ||= ActionScope.new
        end
```

Reply:
> What would be the best and easiest binding to test an action with?

Try the mail binding. You'll have to configure a thing to use, e.g.

```
Thing mail:smtp:local [ hostname="localhost", sender="openhab@yourdomain.com" ]
```

Then you could use $actions.get('mail', 'mail:smtp:local').sendEmail(toaddress, subject, message)

Reply:
I've tested my code, so I can implement either 
```
actions('mail', 'mail:smtp:local').sendEmail(...)
```
or
```
actions['mail']['mail:smtp:local'].sendEmail(...)
```

The first option is simpler in implementation, but the second one isn't too complicated although it involves creating new objects and variables. 

Which one do you prefer, @boc-tothefuture ?

Reply:
I am working on something here. Will be a few but then I can share it for your feedback.